client/AccountsClient.ts

```diff
@@ -62,95 +58,130 @@ export default class AccountsClient {
         return this._redirectClient.init();
     }
 
-    public on(
-        command: RequestType,
-        resolve: (result: RpcResult, state: any) => any,
-        reject?: (error: Error, state: any) => any,
+    public on<T extends RequestType>(
+        command: T,
+        resolve: (result: ResultByRequestType<T>, state: any) => void,
+        reject?: (error: Error, state: any) => void,
     ) {
         this._redirectClient.onResponse(command,
             // State is always an object containing at least the __command property
-            (result: RpcResult, rpcId, state) => resolve(result, state),
-            (error: Error, rpcId, state) => reject && reject(error, state),
+            (result: ResultByRequestType<T>, rpcId, state) => resolve(result, state),
+            (error: Error, rpcId, state) => {
+                if (reject) {
+                    reject(error, state);
+                }
+            },
         );
     }
 
-    public onboard(request: BasicRequest, requestBehavior = this._defaultBehavior): Promise<Account> {
+    public onboard<B extends BehaviorType = DB>(
+        request: BasicRequest,
+        requestBehavior: RequestBehavior<B> = this._defaultBehavior as any,
+    ) {
```

Comment:
Hum, does it really improve the developer experience to remove the return type?
I guess it can be automatically determined now, but the return type is now less obvious by looking at the code.

Reply:
It was not correct before, because it's dependent on the RequestBehavior. Annotating the true conditional type probably doesn't improve dev UX a lot.